### PR TITLE
Modifying the plot option control in Describe>Specific tables/Graphs>Bar Chart

### PR DESCRIPTION
**Pull request by Fidel365**
Fixes(partially) #7556 
@N-thony , @lloyddewit , this PR is ready for preview

### CODE REVIEW

**Comment by lloyddewit** [comment]
@Ivanluv Please could you review?

@Fidel365 Is this PR ready for test by @rdstern or would you prefer that @N-thony tests first?

thanks

**Comment by Fidel365** [comment]
@N-thony can test it first

**Review by Ivanluv — APPROVED** [review]

File: instat/dlgBarAndPieChart.vb [hunk 1]
```diff
@@ -794,21 +746,21 @@ Public Class dlgBarAndPieChart
...
+    '            cmdBarChartOptions.Text = "Bar Chart Options"
+    '        End If
+    '    ElseIf rdoFrequency.Checked Then
+    '        cmdBarChartOptions.Text = "Bar Chart Options"
+    '    ElseIf rdoTreeMap.Checked Then
+    '        cmdBarChartOptions.Text = "Treemap Options"
+    '    ElseIf rdoWordCloud.Checked Then
+    '        cmdBarChartOptions.Text = "Wordcloud Options"
+    '    End If
+    'End Sub
```
Comment (N-thony) [inline instat/dlgBarAndPieChart.vb:763]: remove?

File: instat/dlgBarAndPieChart.vb [hunk 2]
```diff
@@ -882,7 +834,7 @@ Public Class dlgBarAndPieChart
             End If
         End If
         UpdateParameter()
-        ChangeButtonsText()
+        'ChangeButtonsText()
```
Comment (N-thony) [inline instat/dlgBarAndPieChart.vb:837]: remove?

File: instat/dlgBarAndPieChart.vb [hunk 3]
```diff
@@ -1019,4 +972,88 @@ Public Class dlgBarAndPieChart
...
+        End If
+    End Sub
+
+    Private Sub ucrPnlOptions_ControlValueChanged(ucrChangedControl As ucrCore) Handles ucrVariablesAsFactorForBarChart.ControlValueChanged, ucrReceiverX.ControlValueChanged, ucrReceiverLabel.ControlValueChanged, ucrReceiverByFactor.ControlValueChanged, ucrPnlOptions.ControlValueChanged, ucrNudMaxSize.ControlValueChanged, ucrInputReorderX.ControlValueChanged, ucrInputReorderValue.ControlValueChanged, ucrInputAddReorder.ControlValueChanged, ucrChkReorderValue.ControlValueChanged, ucrChkLollipop.ControlValueChanged, ucrChkIncreaseSize.ControlValueChanged, ucrChkAddLabelsText.ControlValueChanged
+        EnableDisableOptions()
+    End Sub
+
+    Private Sub ucrReceiverByFactor_ControlContentsChanged(ucrChangedControl As ucrCore) Handles ucrReceiverByFactor.ControlContentsChanged, ucrPnlOptions.ControlContentsChanged
+
+    End Sub
```
Comment (N-thony) [inline instat/dlgBarAndPieChart.vb:1040]: remove?

File: instat/dlgBarAndPieChart.vb [hunk 4]
```diff
@@ -1019,4 +972,88 @@ Public Class dlgBarAndPieChart
...
+            Me.ucrBase.Location = New Point(7, 413)
+        Else
+            Me.Size = New Size(431, 465)
+            Me.ucrSaveBar.Location = New Point(11, 340)
+            Me.ucrBase.Location = New Point(7, 370)
+        End If
+    End Sub
+    Private Sub ucrReceiverX_ControlContentsChanged(ucrChangedControl As ucrCore) Handles ucrReceiverX.ControlContentsChanged
+
+    End Sub
```
Comment (N-thony) [inline instat/dlgBarAndPieChart.vb:1058]: remove?

**Review by N-thony — CHANGES_REQUESTED** [review]

**Comment by Fidel365** [comment]
@N-thony , I have made the changes requested

File: instat/dlgBarAndPieChart.vb [hunk 5]
```diff
@@ -1019,4 +954,82 @@ Public Class dlgBarAndPieChart
...
+            sdgLayerOptions.SetupLayer(clsNewGgPlot:=clsRggplotFunction, clsNewGeomFunc:=clsGeomTreemapFunction, clsNewGlobalAesFunc:=clsGeomTreemapAesFunction, clsNewLocalAes:=clsLocalRaesFunction, bFixGeom:=True, ucrNewBaseSelector:=ucrBarChartSelector, bApplyAesGlobally:=True, bReset:=bResetBarLayerSubdialog)
+            sdgLayerOptions.ShowDialog()
+            bResetBarLayerSubdialog = False
+        End If
+
+        If rdoWordCloud.Checked Then
+            sdgLayerOptions.SetupLayer(clsNewGgPlot:=clsRggplotFunction, clsNewGeomFunc:=clsGeomTextWordcloudFunction, clsNewGlobalAesFunc:=clsGeomTextWordcloudAesFunction, clsNewLocalAes:=clsLocalRaesFunction, bFixGeom:=True, ucrNewBaseSelector:=ucrBarChartSelector, bApplyAesGlobally:=True, bReset:=bResetBarLayerSubdialog)
+            sdgLayerOptions.ShowDialog()
+            bResetBarLayerSubdialog = False
+        End If
```
Comment (N-thony) [inline instat/dlgBarAndPieChart.vb:1013]: you can use ElseIF on line 1003 and Else on line 1009
also remove line 1007

**Review by N-thony — CHANGES_REQUESTED** [review]
@Fidel365 it will also be good to disable options in the button drop down list when you are not under that option

**Review by rdstern — APPROVED** [review]
@N-thony you had at least one issue with the code.
The new sub-dialogue feature seems to be working, so I am approving.
@lloyddewit perhaps the coding issues can be fixed quickly, and then this could be merged?  Please note there is (I think) another pull request with the lollipop option to merge too.  Perhaps the order of merging needs to be resolved too.
Once they are both merged, I note for the future that I was not able to do a decent pie chart - so I will need to look at that aspect!  I noticed that:
a) I simply couldn't get one nicely!
b) When I flipped the axes it ignored my pie-chart option.  That may be ok, but we should do something about this in the dialogue - perhaps only have the pie chart control visible when the data are not flipped.

**Comment by N-thony** [comment]
@Fidel365 they are open comments above, can you resolve them?

**Comment by rdstern** [comment]
@lloyddewit is this ok to be merged?

**Comment by rdstern** [comment]
@N-thony can someone take this over and resolve the conflicts?  
I think there will be more to do on this dialogue, but it would be good to get these changes merged first.

**Comment by N-thony** [comment]
> @N-thony can someone take this over and resolve the conflicts? I think there will be more to do on this dialogue, but it would be good to get these changes merged first.

yes @derekagorhom is working on it.

**Comment by lloyddewit** [comment]
replaced by PR #8139

**Pull request closed without merging**